Modix.Data.Test/Assertions/DbContextSequenceExtensions.cs

```diff
@@ -85,9 +85,8 @@ public static class DbContextSequenceExtensions
             if (!_valueGeneratorConstructorsByValueType.TryGetValue(property.ClrType, out var valueGeneratorConstructor))
                 throw new ArgumentException($"Unable to generate values of type {property.ClrType.Name}. No suitable {nameof(ValueGenerator)} exists");
 
-            return context.GetService<IValueGeneratorCache>()
-                .GetOrAdd(property, entity, valueGeneratorConstructor)
-                as ResettableSequenceValueGenerator<TProperty>;
+            return (ResettableSequenceValueGenerator<TProperty>)context.GetService<IValueGeneratorCache>()
```

Comment:
There are a few places through the code where I've replaced `as` casts with hard casts. In all of these cases, I've looked through were the results were used, and they're immediately dereferenced. So either it was going to fail with an NRE before, or it will fail with an invalid cast now. And since `as` is definitely nullable, it messes with flow analysis.